zcash_primitives/src/note_encryption.rs

```diff
@@ -438,7 +516,7 @@ pub fn try_sapling_compact_note_decryption(
     plaintext.copy_from_slice(&enc_ciphertext);
     ChaCha20Ietf::xor(key.as_bytes(), &[0u8; 12], 1, &mut plaintext);
 
-    parse_note_plaintext_without_memo(ivk, cmu, &plaintext)
+    parse_note_plaintext_without_memo(ivk, epk, cmu, &plaintext)
 }
 
 /// Recovery of the full note plaintext by the sender.
```

Comment:
this code is repeated in `parse_note_plaintext_without_memo`. try to abstract it into a separate method.